src/include/network/lan/hardware.rb

```diff
@@ -51,19 +51,26 @@ def initialize_network_lan_hardware(include_target)
       Yast.include include_target, "network/lan/cards.rb"
 
       @hardware = nil
+    end
 
-      @widget_descr_hardware = {
-        "HWDIALOG" => {
-          "widget"            => :custom,
-          "custom_widget"     => ReplacePoint(Id(:hw_content), Empty()),
-          "init"              => fun_ref(method(:initHwDialog), "void (string)"),
-          "handle"            => fun_ref(method(:handleHW), "symbol (string, map)"),
-          "store"             => fun_ref(method(:storeHW), "void (string, map)"),
-          "validate_type"     => :function,
-          "validate_function" => fun_ref(method(:validate_hw), "boolean (string, map)"),
-          "help"              => initHelp
-        }
+    def widget_descr_hardware
+      widget_descr = {
+        "widget"        => :custom,
+        "custom_widget" => ReplacePoint(Id(:hw_content), Empty()),
+        "init"          => fun_ref(method(:initHwDialog), "void (string)"),
+        "handle"        => fun_ref(method(:handleHW), "symbol (string, map)"),
+        "store"         => fun_ref(method(:storeHW), "void (string, map)"),
+        "help"          => initHelp
       }
+
+      # validation function currently checks user's input in :ifcfg_name widget
+      # However this widget is present only when adding new device
+      if isNewDevice
```

Comment:
it depends .... from other perspective init method takes dialog definition as provided by ```widget_descr_hardware``` and rebuilds dialog completely according ```isNewDevice```. From this point of view I want to get checks on ```isNewDevice``` out of dialog handlers to be able to split dialog creation completely. In other words I'd like to refactor the code so that dialog is build according ```isNewDevice``` at one place and no additional hacks in handlers are needed. It is doable bcs you know before creating dialog which version you need.